tests/data_processing/test_get_flight_stats.py

```diff
@@ -0,0 +1,139 @@
+        try:
+            pd_testing.assert_frame_equal(a, b)
+        except AssertionError as e:
+            raise self.failureException(msg) from e
+
+    def setUp(self) -> None:
+        self.addTypeEqualityFunc(pd.DataFrame, self.assertDataframeEqual)
+
+    def test_count_flights_structure(self) -> None:
+        self.data_path = 'data/tests/model_test_subset_dataset.json'
+        with open(self.data_path, 'r', encoding='utf-8') as f:
+            self.raw_data = json.load(f)
+        self.airport_data = DataConverter(self.raw_data)
+        airport_df = self.airport_data.airports_df
+        flights_df = self.airport_data.flights
+        flights_df = pd.concat([flights_df, airport_df['icao']], axis=1, join='outer')
+
+        flights_stats = FlightStats(flights_df)
+        # check output dataframe types are int
+        self.assertTrue(all(isinstance(count, int) for count in flights_stats.flight_stats_df['total_arrivals']))
+        self.assertTrue(all(isinstance(count, int) for count in flights_stats.flight_stats_df['total_departures']))
+        # check output shape
+        self.assertEqual(flights_stats.flight_stats_df.shape[1], len(self.FLIGHT_COUNT))
+        # check column names are as expected
+        self.assertEqual(set(flights_stats.flight_stats_df.columns), set(self.FLIGHT_COUNT))
+        # check output is a dataframe
+        self.assertIsInstance(flights_stats.flight_stats_df, pd.DataFrame)
+
+    def test_count_flights_logic_base_case(self) -> None:
+        # base test case
+        fake_input_flights_df = pd.DataFrame(
+            {
+                'icao': 'LBG',
+                'flights': [
+                    [
+                        {'origin_iata': 'LBG'},
+                        {'origin_iata': 'LBG'},
+                        {'origin_iata': 'LBG'},
+                        {'origin_iata': 'SAY'},
+                        {'origin_iata': 'YVR'}
+                    ]
+                ]
+            }
+        )
+        flights_stats = FlightStats(fake_input_flights_df)
+
+        fake_output_flights_df = pd.DataFrame(
+            {
+            'total_arrivals': [2],
+            'total_departures': [3]
+            }
+        )
+        self.assertEqual(flights_stats.flight_stats_df, fake_output_flights_df)
+
+    def test_count_flights_logic_empty_flights(self) -> None:
+        # test output is 0 if the flight lists are empty
+        empty_flights_stats_df = pd.DataFrame(
+            {
+                'icao': 'LBG',
+                'flights':
+                [
+                    [],
+                    [],
+                    [],
+                    []
+                ]
+            }
+        )
+        empty_flights_stats = FlightStats(empty_flights_stats_df)
+        self.assertTrue(all(count == 0 for count in empty_flights_stats.flight_stats_df['total_arrivals']))
+        self.assertTrue(all(count == 0 for count in empty_flights_stats.flight_stats_df['total_departures']))
```

Comment:
Similar to the above comment - I don't understand when the output list for `total_arrivals` and `total_departures ` would ever have more than 1 element. If they wouldn't, could we store them as integers instead?